src/main/java/net/dv8tion/jda/api/utils/messages/MessagePollBuilder.java

```diff
@@ -0,0 +1,271 @@
+ *
+ * @see MessageCreateBuilder#setPoll(MessagePollData)
+ */
+public class MessagePollBuilder
+{
+    private MessagePoll.LayoutType layout = MessagePoll.LayoutType.DEFAULT;
+    private String title;
+    private Map<Long, MessagePoll.Answer> answers = new LinkedHashMap<>();
+    private Duration duration = Duration.ofHours(24);
+    private boolean isMultiAnswer;
+
+    /**
+     * Create a new builder instance
+     *
+     * @param  title
+     *         The poll title (up to {@link MessagePoll#MAX_QUESTION_TEXT_LENGTH} characters)
+     *
+     * @throws IllegalArgumentException
+     *         If the title is blank or longer than {@link MessagePoll#MAX_QUESTION_TEXT_LENGTH} characters
+     */
+    public MessagePollBuilder(@Nonnull String title)
+    {
+        this.setTitle(title);
+    }
+
+    /**
+     * They poll layout.
+     *
+     * @param  layout
+     *         The layout
+     *
+     * @throws IllegalArgumentException
+     *         If null or {@link net.dv8tion.jda.api.entities.messages.MessagePoll.LayoutType#UNKNOWN UNKNOWN} is provided
+     *
+     * @return The updated builder
+     */
+    @Nonnull
+    public MessagePollBuilder setLayout(@Nonnull MessagePoll.LayoutType layout)
+    {
+        Checks.notNull(layout, "Layout");
+        Checks.check(layout != MessagePoll.LayoutType.UNKNOWN, "Layout cannot be UNKNOWN");
+
+        this.layout = layout;
+        return this;
+    }
+
+    /**
+     * Change the title for this poll.
+     *
+     * @param  title
+     *         The poll title (up to {@link MessagePoll#MAX_QUESTION_TEXT_LENGTH} characters)
+     *
+     * @throws IllegalArgumentException
+     *         If the title is blank or longer than {@link MessagePoll#MAX_QUESTION_TEXT_LENGTH} characters
+     *
+     * @return The updated builder
+     */
+    @Nonnull
+    public MessagePollBuilder setTitle(@Nonnull String title)
+    {
+        Checks.notBlank(title, "Title");
+        title = title.trim();
+        Checks.notLonger(title, MessagePoll.MAX_QUESTION_TEXT_LENGTH, "Poll question title");
+
+        this.title = title;
+        return this;
+    }
+
+    /**
+     * Change the duration for this poll.
+     * <br>Default: {@code 1} day
+     *
+     * <p>The poll will automatically expire after this duration.
+     *
+     * @param  duration
+     *         The duration of this poll (in hours resolution)
+     *
+     * @throws IllegalArgumentException
+     *         If the duration is null, less than 1 hour, or longer than {@value  MessagePoll#MAX_DURATION_HOURS} hours (7 days)
+     *
+     * @return The updated builder
+     */
+    @Nonnull
+    public MessagePollBuilder setDuration(@Nonnull Duration duration)
+    {
+        Checks.notNull(duration, "Duration");
+        Checks.positive(duration.toHours(), "Duration");
+        Checks.check(duration.toHours() <= MessagePoll.MAX_DURATION_HOURS, "Poll duration may not be longer than 7 days. Provided: %d hours", duration.toHours());
```

Comment:
I feel like this text can be confusing when it speaks about text, but then gives Hours... Maybe have either the max hours mentioned, or display duration as days?

Reply:
I don't understand whats confusing here?

Reply:
I mentioned it.
It say can't be higher than 7 days, but returns your input in hours.

Reply:
Maybe "Poll duration may not be longer than 168 hours (7 days). Provided: %d hours" or something similar.
I agree with @Andre601, its a bit confusing to give a limit in a different time unit than what youre actually providing.